### PR TITLE
add aws auth support

### PR DESCRIPTION
**Pull request by 4killo**
- add aws auth support

### CODE REVIEW

File: vault/vault.go [hunk 1]
```diff
@@ -40,30 +49,35 @@ type ClientCredential struct {
 type Configuration struct {
 	Credentials              ClientCredential
 	Tenant, TLD, URLTemplate string
+	Provider                 auth.Provider
 }
 
-// Vault provides access to secrets stored in Delinea DSV
+// Vault provides access to secrets stored in Thycotic DSV
```
Comment (andrii-zakurenyi) [inline vault/vault.go:55]: Probably this change should be reverted :)

Comment (4killo) [reply vault/vault.go:55]: good catch!

**Comment by andrii-zakurenyi** [comment]
Approved :+1: (I do not have permissions to approve it using button).

**Comment by mariiatuzovska** [comment]
:+1:

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -20,10 +20,10 @@ jobs:
     name: Build
     runs-on: ubuntu-latest
     steps:
-      - name: Set up Go 1.13
+      - name: Set up Go 1.18
         uses: actions/setup-go@v1
         with:
-          go-version: 1.13
+          go-version: 1.18
```
Comment (tdillenbeck) [inline .github/workflows/tests.yml:26]: Yay! 👍

File: auth/aws_test.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+func TestGetSTSHeaderAndBody(t *testing.T) {
+
+	testCases := []struct {
+		name              string
+		getCallerIdentity func() *request.Request
+		expectedHeader    string
+		expectedBody      string
+		expectedError     error
+	}{
+		{"good Response", func() *request.Request {
```
Comment (tdillenbeck) [inline auth/aws_test.go:22]: question: Are there other tests we need other than the happy path?

**Review by tdillenbeck — APPROVED** [review]

File: auth/aws_test.go (hunk 3 above)
Comment (4killo) [reply auth/aws_test.go:22]: Good Catch! We will fix this next GCP and AZUR Pr

**Pull request merged**